app/streaming/trace/service/service.go

```diff
@@ -0,0 +1,87 @@
+package service
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/gogo/protobuf/proto"
+	"github.com/provenance-io/provenance/internal/streaming"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+var _ streaming.StreamService = (*TraceStreamingService)(nil)
+
+// TraceStreamingService is a lightweight streaming service for showing how streaming is plugged in.
+type TraceStreamingService struct {
+	printDataToStdout bool              // Print request response data to stdout.
+	codec             codec.BinaryCodec // binary marshaller used for re-marshaling the ABCI messages to write them out to the destination files
+}
+
+func NewTraceStreamingService(
+	printDataToStdout bool,
+	marshaller codec.BinaryCodec,
+) *TraceStreamingService {
+	tss := &TraceStreamingService{
+		printDataToStdout: printDataToStdout,
+		codec:             marshaller,
+	}
+
+	return tss
+}
+
+// StreamBeginBlocker writes out the received BeginBlockEvent request and response
+func (tss *TraceStreamingService) StreamBeginBlocker(
+	ctx sdk.Context,
+	req abci.RequestBeginBlock,
+	res abci.ResponseBeginBlock,
+) {
+	errMsg := "BeginBlocker listening hook failed"
+
+	// write req
+	if err := tss.write(ctx, &req); err != nil {
+		ctx.Logger().Error(errMsg, "height", req.Header.Height, "err", err)
```

Comment:
There needs to be something here and in the other one that allows someone to look at the log line and distinguish between the two errors. Maybe add the args `, "action", "write request"` and `, "action", "write response"` to each log statement respectively.

Same goes for the two log messages in the endblocker listener hook.